libraries/botbuilder-runtime-integration-express/src/index.ts

```diff
@@ -40,44 +41,30 @@ export async function start(
     settingsDirectory: string,
     options: Partial<Options> = {}
 ): Promise<void> {
-    const { port, messagingEndpointPath } = TypedOptions.check(Object.assign({}, defaultOptions, options));
     const [services, configuration] = await getRuntimeServices(applicationRoot, settingsDirectory);
+    const [_, listen] = await makeApp(services, configuration, options);
 
-    const app = await makeApp(messagingEndpointPath, services, configuration);
-    const { bot } = await services.mustMakeInstances('bot');
-
-    // The 'upgrade' event handler for processing WebSocket requests needs to be registered on the Node.js http.Server, not the Express.Application.
-    // In Express the underlying http.Server is made available after the app starts listening for requests.
-    const server = app.listen(port, () => {
-        console.log(`server listening on port ${port}`);
-    });
-
-    server.on('upgrade', async (req, socket, head) => {
-        const adapter = await services.mustMakeInstance('adapter');
-        adapter.useWebSocket(req, socket, head, async (context) => {
-            await bot.run(context);
-        });
-    });
+    listen();
 }
 
 /**
  * Create an Express App using the runtime Express integration.
  *
- * @param messagingEndpointPath path for receiving and processing [Activities](xref:botframework-schema.Activity)
  * @param services runtime service collection
  * @param configuration runtime configuration
- * @returns an Express App ready to listen for connections
+ * @param options options bag for configuring Express Application
+ * @returns the Express Application and a function to start the App & handle "upgrade" requests for Streaming
  */
 export async function makeApp(
-    messagingEndpointPath: string,
     services: ServiceCollection<IServices>,
-    configuration: Configuration
-): Promise<express.Application> {
+    configuration: Configuration,
+    options: Partial<Options> = {}
+): Promise<[express.Application, (callback?: () => void) => http.Server]> {
```

Comment:
Typescript now supports labeling tuple fields. This could be:

```typescript
Promise<[app: express.Application, listen: (callback?: () => void) => http.Server]> { ... }
```

Which just provides a touch more context, implicitly, to callers of the function:
<img width="1042" alt="Screen Shot 2021-03-08 at 1 56 47 PM" src="https://user-images.githubusercontent.com/1235378/110387204-226aea00-8016-11eb-8791-0bdc5bebe0b6.png">

Reply:
Good call, done.